사이트 유지보수

rkttu opened the pull request:
다음의 사항들을 업데이트했습니다.

- 미디엄 블로그로 가는 메인 메뉴 링크 추가
- 미디엄 블로그와 겹치는 포스트 제거
- 오래된 공지 배너 삭제

rkttu committed:
Modify menu setup

rkttu committed:
Remove old banner

rkttu committed:
Remove duplicated post

sukso96100 commented:
![image](https://user-images.githubusercontent.com/1916739/101635973-059e3500-3a6e-11eb-9e47-1df5cf498b01.png)
메인화면 배너쪽 관련해서 코드 수정이 필요해 보입니다. 설정 파일에서 설정값이 비어 있으면 렌더링 하지 않도록 하는 식으로 템플릿 수정이 필요할 듯 합니다.

sukso96100 commented on content/_index.md:
```diff
@@ -6,11 +6,6 @@ note: "커뮤니티에서 활동하려면, 커뮤니티 규정에 동의해야 
 buttons:
     titles: ["WSL 시작하기", "Facebook Group", "GitHub"]
     links: ["https://wslhub.com/wsl-firststep", "https://www.facebook.com/groups/wslhub", "https://github.com/wslhub"]
-banner:
-    text: "W$L/CONF>_ mini 컨퍼런스가 2020.11.11 오후 8시에 예정되어 있습니다."
-    button: 
-        text: "참여하기"
-        link: "https://www.notion.so/W-L-CONF-_-mini-20-11-9e0956e39e644f529c63776057dd21e9"
```
나중에 배너 사용할 일 있을 때 고려해서, 이쪽은 완전히 지우는 것 보단 값만 비워두고 키는 남겨두는것이 어떨까요?

rkttu replied:
네. 앞서 말씀해주신 내용을 감안했을 때, 이렇게 해보면 어떨까요?

- 템플릿 언어 상에서 banner 블록이 정의가 되어있는지 확인 (null check)해서 null이면 렌더링 생략. 아니면 렌더링.
- _index.md의 banner 아랫 부분은 주석처리

sukso96100 replied:
`banner`부분 주석처리 하는것도 좋은 것 같습니다. 👍

sukso96100 reviewed (changes requested):
커멘트 확인해주시면 감사하겠습니다. :)

sukso96100 commented:
> ![image](https://user-images.githubusercontent.com/1916739/101635973-059e3500-3a6e-11eb-9e47-1df5cf498b01.png)
> 메인화면 배너쪽 관련해서 코드 수정이 필요해 보입니다. 설정 파일에서 설정값이 비어 있으면 렌더링 하지 않도록 하는 식으로 템플릿 수정이 필요할 듯 합니다.

이쪽에 `div` 요소를 배너 관련 설정값 비어있는지 여부로 [조건부 렌더링](https://gohugo.io/templates/introduction/#conditionals)을 하면 됩니다.
https://github.com/wslhub/wslhub.github.io/blob/e8d58d92e4b40c0776b7a2eb4a047059b112c993/layouts/index.html#L19-L22

rkttu committed:
Implement banner display toggle

- To display a banner, uncomment and
add banner blocks on content/_index.md front matter
- To turn off, remove or comment banner data block.

rkttu commented:
> > ![image](https://user-images.githubusercontent.com/1916739/101635973-059e3500-3a6e-11eb-9e47-1df5cf498b01.png)
> > 메인화면 배너쪽 관련해서 코드 수정이 필요해 보입니다. 설정 파일에서 설정값이 비어 있으면 렌더링 하지 않도록 하는 식으로 템플릿 수정이 필요할 듯 합니다.
> 
> 이쪽에 `div` 요소를 배너 관련 설정값 비어있는지 여부로 [조건부 렌더링](https://gohugo.io/templates/introduction/#conditionals)을 하면 됩니다.
> https://github.com/wslhub/wslhub.github.io/blob/e8d58d92e4b40c0776b7a2eb4a047059b112c993/layouts/index.html#L19-L22

관련 커밋을 하나 넣었습니다.

sukso96100 commented:
![image](https://user-images.githubusercontent.com/1916739/101637276-e43e4880-3a6f-11eb-9083-cc0f2a9b1426.png)
이제 잘 작동합니다. 승인 후 병합하도록 하겠습니다. 감사합니다. 👍
